modules/flyte/identities.tf

```diff
@@ -0,0 +1,40 @@
+resource "azurerm_user_assigned_identity" "flyte_controlplane" {
```

Comment:
Nitpick (non blocking),regarding terraform DRY principles. it seems that all of this could just be (⚠️ Not tested)

```hcl
locals {
    mapping = {
      flytepropeller = "flyte_controlplane"
      flyteadmin = "flyte_dataplane"
      datacatalog = "flyte_controlplane"
  }
}

resource "azurerm_user_assigned_identity" "this" {
  for_each            = toset(values(local.mapping))
  name                = each.key
  location            = var.azurerm_resource_group_location
  resource_group_name = var.azurerm_resource_group_name
  tags                = var.tags
}


resource "azurerm_federated_identity_credential" "this" {
  name                = toset(keys(local.mapping))
  name                = each.key
  resource_group_name = var.azurerm_resource_group_name
  audience            = ["api://AzureADTokenExchange"]
  issuer              = var.azurerm_kubernetes_cluster_oidc_issuer_url
  parent_id           = azurerm_user_assigned_identity.this[local.mapping[each.key]].id
  subject             = "system:serviceaccount:${var.namespaces.platform}:${var.serviceaccount_names[each.key]}"
}

```

Reply:
Great suggestion!

Reply:
The code is cleaner but the only downside is that the generated README.md now shows a `this` resource for `azurerm_user_assigned_identity` and `azurerm_federated_identity_credential` instead of the actual resources.